api/go.mod

```diff
@@ -3,7 +3,7 @@
 
 module github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/api
 
-go 1.21
+go 1.22
```

Comment:
Curious, is the Go bump needed here?

Reply:
Yes, this was needed to ensure go 1.22 is used across the project. Having it set to 1.21 caused inconsistent language versions to be used, for example for loop variables scoping was still in 1.21 mode in the test i wrote here, which led to issues that I couldn't figure out for a while.

Thinking about this again, i think I could not bump these here and instead use `//go:build` directives in the necessary test files, but I think that is more opaque so would prefer to keep it consistent at go 1.22 in go mods.

Let me know if you can see any problems with that.